### PR TITLE
[flutter_tools] do not require a dependency on devtools server

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Re-arrange the implementation of the devtools launcher so that google3 is not required to depend on any devtools packages. Also renames the `build_runner` folders to `isolated` to better clarify their intention.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Not sure if this would have prevented the recent build breakage?

File: packages/flutter_tools/lib/executable.dart [hunk 1]
```diff
@@ -51,6 +45,13 @@ import 'src/commands/upgrade.dart';
 import 'src/commands/version.dart';
 import 'src/features.dart';
 import 'src/globals.dart' as globals;
+// Files in `isolated` are intentionally excluded from g3 sync.
+import 'src/isolated/devtools_launcher.dart';
+import 'src/isolated/mustache_template.dart';
+import 'src/isolated/resident_web_runner.dart';
+import 'src/isolated/web_compilation_delegate.dart';
+//
```
Comment (chingjun) [inline packages/flutter_tools/lib/executable.dart:53]: Nit: remove the extra `//`

**Review by chingjun — APPROVED** [review]
LGTM! Thanks!

Looks like it would indeed prevent the breakages if `isolated` is actually, well, isolated :)

File: packages/flutter_tools/lib/executable.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/executable.dart]: Done

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 2]
```diff
@@ -1683,43 +1682,15 @@ String nextPlatform(String currentPlatform, FeatureFlags featureFlags) {
...
-        observatoryAddress,
-        'http://${_devtoolsServer.address.host}:${_devtoolsServer.port}',
-        false,  // headless mode,
-        false,  // machine mode
-      );
-    } on Exception catch (e, st) {
-      globals.printTrace('Failed to launch DevTools: $e\n$st');
-    }
-  }
+/// An launcher for the devtools debugger and analysis tool.
```
Comment (helin24) [inline packages/flutter_tools/lib/src/resident_runner.dart:1685]: a launcher

**Review by helin24 — APPROVED** [review]

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Done

**Comment by jonahwilliams** [comment]
Landed the internal CL to exclude the isolated/ folder. @renyou can frob handle changes to the BUILD file, or will this PR necessarily break the roll?

**Comment by jonahwilliams** [comment]
G3 fix applied, waiting for results

**Pull request merged**